src/main/java/com/conveyal/taui/controllers/WorkerController.java

```diff
@@ -185,18 +192,83 @@ private String jsonResponse (Response response, int statusCode, Object object) {
         }
     }
 
+    private static class ExtendedJobStatus extends JobStatus {
+        public RegionalAnalysis regionalAnalysis;
+
+        public ExtendedJobStatus(JobStatus js) {
+            super();
+            this.jobId = js.jobId;
+            this.graphId = js.graphId;
+            this.workerCommit = js.workerCommit;
+            this.total = js.total;
+            this.complete = js.complete;
+            this.incomplete = js.incomplete;
+            this.deliveries = js.deliveries;
+            this.deliveryPass = js.deliveryPass;
+        }
+    }
+
     /**
      * Fetch status of all unfinished jobs as a JSON list.
      */
     private String getAllJobs(Request request, Response response) {
-        return jsonResponse(response, HttpStatus.OK_200, broker.getJobSummary());
+        String accessGroup = request.attribute("accessGroup");
+        if (!AnalysisServerConfig.adminAccessGroup.equals(accessGroup)) {
+            response.status(401);
+            return "You do not have access.";
+        }
+
+        List<ExtendedJobStatus> extendedJobStatuses = new ArrayList<>();
```

Comment:
I would expect the logic to fill in the fields of an ExtendedJobStatus to be encapsulated within that class, either in its constructor, or if calling out to a database in a constructor is deemed problematic, in an instance method on the class. I know that there are various places in R5 we're operating directly on the fields of classes from other classes, and there's a balance to be struck between encapsulation and simplicity / readability. But in cases like this it's pretty easy to group the code with the class it acts upon.